bgpd/bgp_route.c

```diff
@@ -5478,6 +5478,9 @@ static int bgp_aggregate_info_same(struct bgp_info *ri, uint8_t origin,
 	if (!community_cmp(ri->attr->community, comm))
 		return 0;
 
+	if (!CHECK_FLAG(ri->flags, BGP_INFO_VALID))
```

Comment:
Don,

Some questions for my understanding.

Does the scenario involve issuing a aggregate-command, then issuing a "no aggregate-command" and issuing the aggregate-command for the 2nd time ? Is that the scenario that you were mentioning ?

I have few followup questions based on this.

Reply:
Not quite.  When you enter the same command a second time, it currently does an "unset" and then "set" to pick up any parameter changes (summary-only, as-set).  The problem is triggered because when the "set" part of the action happens, the original aggregate that has been marked for removal but is still around.   Prior to this change, it would consider the one being removed identical to the new one and not do the new install.  Since the prior was marked for removal, it would be removed from the bgp table at a later time, leaving no aggregate entry.

Reply:
Got it. This will guard against the case where "aggregate-command" is issued two times in short span of time. 

Curious to see what happens if user issues "no aggregate-command" twice in short span of time ?

Also moving this check to the beginning of this function may be better, as once this new condition matches irrespective of what other conditions are matched or not, we still want to go ahead and return '0'

Reply:
I tested the case of two immediate deletions as well, and it works properly.  The second one receives a message of "% There is no aggregate-address configuration." and the aggregate is removed and component routes unsuppressed.  I'm fine moving the check to the beginning of the function and will make the change.

Reply:
Actually, I really don't know which of the exit criteria is more likely to be hit in this function, so moving this particular test to the beginning doesn't seem necessary.  If you have a reason to think it would better to move it to the beginning, I'm willing to do it, just not sure it's needed or useful.